beacon-chain/blockchain/core.go

```diff
@@ -85,7 +85,19 @@ func NewBeaconChain(db ethdb.Database) (*BeaconChain, error) {
 
 // GenesisBlock returns the canonical, genesis block.
 func (b *BeaconChain) GenesisBlock() (*types.Block, error) {
-	return types.NewGenesisBlock()
+	genesisExists, err := b.db.Has([]byte("genesis"))
+	if err != nil {
+		return nil, err
+	}
+	if genesisExists {
+		bytes, _ := b.db.Get([]byte("genesis"))
```

Comment:
Please don’t dump this error. All errors must be handled.

Reply:
Error handled 👍